src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -47,6 +49,14 @@
 import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Component;
 
+/**
+ * SamService encapsulates logic for interacting with Sam. HTTP Statuses returned by Sam are
+ * interpreted by the functions in this class.
+ *
+ * <p>This class is used both by Flights and outside of Flights. Within flights,
+ * InterruptedExceptions need to be raised and handled by Stairway. Outside of flights, they are
+ * unexpected and will be converted into {@link SamInterruptedException} which are unchecked.
```

Comment:
nit: update the comment on what the right way to do this is?

Reply:
Done.